src/libinjector/linux/linux_syscalls.c

```diff
@@ -219,6 +219,115 @@ bool setup_read_syscall(injector_t injector, x86_registers_t* regs, int fd, addr
     return true;
 }
 
+bool setup_exit_syscall(injector_t injector, x86_registers_t* regs, uint64_t rc)
+{
+    struct argument args[1] = { {0} };
+    init_int_argument(&args[0], rc);
+
+    regs->rax = sys_exit;
+    regs->rip = injector->syscall_addr;
+
+    if (!setup_stack(injector->drakvuf, regs, args, ARRAY_SIZE(args)))
+    {
+        PRINT_DEBUG("Failed to setup read syscall\n");
+        return false;
+    }
+    return true;
+}
+
+bool setup_execve_syscall(injector_t injector, x86_registers_t* regs, addr_t virtual_memory_addr)
```

Comment:
This function should just setup the `execve` syscall. Can we move everything related to creating an empty "argv" and "envp" to `linux_exec_file.c`? Also, this function is quite long. Can you for example extract `place_string_array_on_addr` function? It looks like it can be used for creating `envp[]` and `argv[]`.
```suggestion
bool setup_execve_syscall(injector_t injector, x86_registers_t* regs, addr_t pathname, addr_t argv, addr_t envp)
```

Reply:
I will try to do it

Reply:
I have created a #1303 for adding support for arrays, please have a look at it